src/main/java/org/springframework/data/jpa/repository/query/QueryUtils.java

```diff
@@ -205,12 +50,11 @@ private QueryUtils() {
 	 */
 	public static String getExistsQueryString(String entityName, String countQueryPlaceHolder,
 			Iterable<String> idAttributes) {
-
-		String whereClause = Streamable.of(idAttributes).stream() //
-				.map(idAttribute -> String.format(EQUALS_CONDITION_STRING, "x", idAttribute, idAttribute)) //
-				.collect(Collectors.joining(" AND ", " WHERE ", ""));
-
-		return String.format(COUNT_QUERY_STRING, countQueryPlaceHolder, entityName) + whereClause;
+		if (JSQLPARSER_IN_CLASSPATH) {
```

Comment:
I cannot really follow what you suggest. The problem is that QueryUtils is never initialised (since util class) and we only know what option (JSql or custom parser) we should use at runtime. Additionally a given request can be parsed with JSql and the next not.

Reply:
What I could do is to make `JSqlParserQueryUtils` and `DefaultQueryUtils` instantiateable and then have a factory that creates the right Utils utils class based on the provided query.

Reply:
Yes, that is what I proposed. @mp911de had a somewhat different idea though. I asked him to provide a review as well. Let's see what comes out of that.

Reply:
Ok thanks! I am looking forward to it! 👍